engine/src/test/java/io/camunda/zeebe/engine/ArchitectureTest.java

```diff
@@ -1,37 +0,0 @@
-/*
- * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
- * one or more contributor license agreements. See the NOTICE file distributed
- * with this work for additional information regarding copyright ownership.
- * Licensed under the Zeebe Community License 1.1. You may not use this file
- * except in compliance with the Zeebe Community License 1.1.
- */
-package io.camunda.zeebe.engine;
-
-import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.noClasses;
-
-import com.tngtech.archunit.core.importer.ImportOption;
-import com.tngtech.archunit.junit.AnalyzeClasses;
-import com.tngtech.archunit.junit.ArchTest;
-import com.tngtech.archunit.lang.ArchRule;
-
-@AnalyzeClasses(packages = "io.camunda.zeebe", importOptions = ImportOption.DoNotIncludeTests.class)
-public class ArchitectureTest {
-
-  @ArchTest
-  public static final ArchRule RULE_ENGINE_CLASSES_MUST_NOT_DEPEND_ON_STREAMPROCESSOR_PACKAGE =
```

Comment:
:thought_balloon:  @Zelldon As long as the stream package isn't split into api and impl, this architecture test still makes sense IMO.

Reply:
I guess than we have to adjust it to the new names.

Reply:
I removed the last commit. The package names were already adjusted.

Reply:
So one issue we have now is that `NextValueManager` is used in the Engine as well. We can maybe exclude that or duplicate the code since it is not that complicate (I really dislike to write such thing) but I see no other way. We can't move it to zb-db since it would introduce dependency to msgpack as well.

Reply:
i see no way to exclude a class in the test

Reply:
https://github.com/camunda/zeebe/pull/10979/commits/61374a228e4d93608d84bc55001bcc68bd9ac568 and follow up https://github.com/camunda/zeebe/issues/10980

Reply:
Splitting/duplicating is reasonable IMO, there's no requirement that we use the same code for determining the next process version and getting the next key :+1: